mlflow/gateway/app.py

```diff
@@ -109,34 +90,52 @@ def _route_type_to_endpoint(config: RouteConfig):
     )
 
 
-def _add_dynamic_route(route_config: RouteConfig):
-    app.add_api_route(
-        path=f"/gateway/routes/{route_config.name}",
-        endpoint=_route_type_to_endpoint(route_config),
-        methods=["POST"],
+def create_app_from_config(config: GatewayConfig) -> GatewayApp:
+    """
+    Creates an MLflow Gateway API app.
+    """
+    app = GatewayApp(
+        config=config,
+        title="MLflow Gateway API",
+        description="The core gateway API for reverse proxy interface using remote inference "
+        "endpoints within MLflow",
+        version=VERSION,
     )
-    ACTIVE_ROUTES.append(route_config.to_route())
 
+    @app.get("/")
+    async def index():
+        return RedirectResponse(url="/docs")
 
-def _add_routes(routes: List[RouteConfig]):
-    for route in routes:
-        _add_dynamic_route(route)
+    @app.get(MLFLOW_GATEWAY_HEALTH_ENDPOINT)
+    async def health():
+        return {"status": "OK"}
 
+    @app.get("/gateway/routes/{route_name}")
+    async def get_route(route_name: str):
+        if matched := app.get_dynamic_route(route_name):
+            return {"route": matched}
 
-def create_app(gateway_conf_path: Union[str, Path]) -> FastAPI:
-    """
-    Create the FastAPI app by loading the dynamic route configuration file from the
-    specified local path and generating POST methods
-    """
-    route_config = _load_route_config(gateway_conf_path)
-    _add_routes(route_config.routes)
-    return app
```

Comment:
The original `create_app` was like a mutator. Every time we call it, it add new routes, which makes it difficult to test the app or separate tests.